aptos-move/move-examples/fungible_asset/stablecoin/sources/usdk.move

```diff
@@ -0,0 +1,333 @@
+    const EPAUSED: u64 = 2;
+    /// The account is already a minter
+    const EALREADY_MINTER: u64 = 3;
+    /// The account is not a minter
+    const ENOT_MINTER: u64 = 4;
+    /// The account is denylisted
+    const EDENYLISTED: u64 = 5;
+
+    const ASSET_SYMBOL: vector<u8> = b"USDK";
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    struct Roles has key {
+        master_minter: address,
+        minters: vector<address>,
+        pauser: address,
+        denylister: address,
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    struct Management has key {
+        extend_ref: ExtendRef,
+        mint_ref: MintRef,
+        burn_ref: BurnRef,
+        transfer_ref: TransferRef,
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    struct State has key {
+        paused: bool,
+    }
+
+    struct Approval has drop {
+        owner: address,
+        to: address,
+        nonce: u64,
+        chain_id: u8,
+        spender: address,
+        amount: u64,
+    }
+
+    #[event]
+    struct Mint has drop, store {
+        minter: address,
+        to: address,
+        amount: u64,
+    }
+
+    #[event]
+    struct Burn has drop, store {
+        minter: address,
+        from: address,
+        store: Object<FungibleStore>,
+        amount: u64,
+    }
+
+    #[event]
+    struct Pause has drop, store {
+        pauser: address,
+        is_paused: bool,
+    }
+
+    #[event]
+    struct Denylist has drop, store {
+        denylister: address,
+        account: address,
+    }
+
+    #[view]
+    public fun usdk_address(): address {
+        object::create_object_address(&@stablecoin, ASSET_SYMBOL)
+    }
+
+    #[view]
+    public fun metadata(): Object<Metadata> {
+        object::address_to_object(usdk_address())
+    }
+
+    /// Called as part of deployment to initialize the stablecoin.
+    fun init_module(usdk_signer: &signer) {
+        // Create the stablecoin with primary store support.
+        let constructor_ref = &object::create_named_object(usdk_signer, ASSET_SYMBOL);
+        primary_fungible_store::create_primary_store_enabled_fungible_asset(
+            constructor_ref,
+            option::none(),
+            utf8(ASSET_SYMBOL), /* name */
+            utf8(ASSET_SYMBOL), /* symbol */
+            8, /* decimals */
+            utf8(b"http://example.com/favicon.ico"), /* icon */
+            utf8(b"http://example.com"), /* project */
+        );
+
+        // Set ALL stores for the fungible asset to untransferable.
+        fungible_asset::set_untransferable(constructor_ref);
+
+        // All resources created will be kept in the asset metadata object.
+        let metadata_object_signer = &object::generate_signer(constructor_ref);
+        move_to(metadata_object_signer, Roles {
+            master_minter: @master_minter,
+            minters: vector::empty(),
```

Comment:
nit: Use vector[] here instead

Reply:
Also why changed this from vector[@minter]. I still see the minter address in Move.toml

Reply:
Wanted to test with the add_minter function in unit tests